### PR TITLE
Mole knight

### PR DESCRIPTION
**Pull request by Mach131**

### CODE REVIEW

File: data/card_definitions.json [hunk 1]
```diff
@@ -10457,6 +10457,324 @@
...
+		"boost": {
+			"boost_type": "immediate",
+			"force_cost": 0,
+			"cancel_cost": -1,
+			"display_name": "Lost Treasure",
+			"effects": [
+				{
+					"timing": "immediate",
+					"effect_type": "choose_discard",
+					"limitation": "special/ultra",
```
Comment (daniel-k-taylor) [inline data/card_definitions.json:10585]: I probably would have done special_or_ultra, but this is fine i guess.

File: data/card_definitions.json [hunk 2]
```diff
@@ -10457,6 +10457,324 @@
...
+			"display_name": "Come At Me!",
+			"effects": [
+				{
+					"timing": "during_strike",
+					"effect_type": "powerup",
+					"amount": 3
+				},
+				{
+					"timing": "start_of_next_turn",
+					"effect_type": "add_to_gauge_immediately"
```
Comment (daniel-k-taylor) [inline data/card_definitions.json:10646]: This should have "and draw a card"  too

File: data/decks/mole.json [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+		}
+	}],
+	"ability_effects": [
+		{
+			"timing": "set_strike",
+			"condition": "initiated_strike",
+			"effect_type": "choice",
+			"choice": [
+				{
+					"effect_type": "set_used_character_bonus",
```
Comment (daniel-k-taylor) [inline data/decks/mole.json:46]: Wouldn't add_attack_effect work? Like Kokonoe?

File: data/decks/mole.json [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+					"character_effect": true,
+					"linked_effect": {
+						"timing": "before",
+						"condition": "buddy_space_unoccupied",
+						"condition_detail": "Burrow",
+						"effect_type": "move_to_buddy",
```
Comment (daniel-k-taylor) [inline data/decks/mole.json:85]: how does this interact with cannot move effects?
The text says "If you did, +2 power"
Did what? Tried to advance or actually advanced into the burrow?

File: data/card_definitions.json (hunk 1 above)
Comment (daniel-k-taylor) [reply data/card_definitions.json:10585]: in fact, yours probably works better if you want to just use the string as is.

**Review by daniel-k-taylor — CHANGES_REQUESTED** [review]

File: data/card_definitions.json (hunk 1 above)
Comment (Mach131) [reply data/card_definitions.json:10585]: yeah, mostly went with this for the string appearance

File: data/card_definitions.json (hunk 2 above)
Comment (Mach131) [reply data/card_definitions.json:10646]: oh huh, this version is better than the S6 version

File: data/decks/mole.json (hunk 3 above)
Comment (Mach131) [reply data/decks/mole.json:46]: Should be the same thing; i guess it avoids the need to repeat the effect

File: data/decks/mole.json (hunk 4 above)
Comment (Mach131) [reply data/decks/mole.json:85]: I checked, you only get the power if you successfully advance/retreat into the burrow (hence the on_buddy_space check)

**Review by daniel-k-taylor — APPROVED** [review]

**Pull request merged**